sdks/typescript/src/opik/tracer/Trace.ts

```diff
@@ -0,0 +1,61 @@
+import { OpikClient } from "@/client/Client";
+import type {
+  Span as ISpan,
+  Trace as ITrace,
+  TraceUpdate,
+} from "@/rest_api/api";
+import { v7 as uuid } from "uuid";
+import { SavedSpan, Span } from "./Span";
+
+export interface SavedTrace extends ITrace {
+  id: string;
+}
+
+interface SpanData extends Omit<ISpan, "startTime" | "traceId"> {
+  startTime?: Date;
+}
+
+export class Trace {
+  private spans: Span[] = [];
+
+  constructor(
+    private data: SavedTrace,
+    private opik: OpikClient
+  ) {}
+
+  public end = async () => {
+    await this.update({ endTime: new Date() });
+  };
+
+  public span = async (spanData: SpanData) => {
+    const projectName = spanData.projectName ?? this.opik.config.projectName;
+    const spanWithId: SavedSpan = {
+      id: uuid(),
+      startTime: new Date(),
```

Comment:
we should consider having `utc` for all dates probably

Reply:
i would also add a way to get the time from arguments and if it is not specified, i would it the current one

Reply:
Well, the spread operator acts like that:
```typescript
{
  <default values>,
  ...argumentValues,
  <forced values>
}```

In this case `id` and `startTime` have default values and `projectName` and `traceId` are forced values

I think the structure of it is correct but I could get the defaults in another way more centralized if that's what you mean :)

Reply:
yeah, sorry, misread it